src/components/CombinedFeaturePanel/components/Gallery/Gallery.tsx

```diff
@@ -1,64 +1,185 @@
+  const [activeIndex, setActiveIndex] = useState(-1);
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  const activeImage = images[activeIndex];
+  const next = () => {
+    setActiveIndex(activeIndex + 1 >= images.length ? 0 : activeIndex + 1);
+  };
+  const previous = () => {
+    setActiveIndex(activeIndex - 1 < 0 ? images.length - 1 : activeIndex - 1);
+  };
+  const goTo = (imageId: string) => {
+    setActiveIndex(getImageIndex(imageId));
+    setDialogOpen(true);
+  };
+  const close = () => {
+    setDialogOpen(false);
+  };
+  const getDetailPageUrl = (imageId: string) => {
+    return `${baseFeatureUrl}/images/${imageId}`;
+  };
+  const getReportUrl = (imageId: string) => {
+    return `${getDetailPageUrl(imageId)}/report`;
+  };
+
+  const api: GalleryContextType = {
+    activeImage,
+    activeIndex,
+    size: images.length,
+    next,
+    goTo,
+    close,
+    previous,
+    getDetailPageUrl,
+    getReportUrl,
+  };
+
+  // Open the dialog on initial page load when visiting the /images/[imageId] page
+  // biome-ignore lint/correctness/useExhaustiveDependencies:
+  useEffect(() => {
+    if (activeImageId) goTo(activeImageId);
+  }, [images, activeImageId]);
 
-  > .wrapper {
-    position: absolute;
-    width: 100%;
-    z-index: 1;
-    bottom: -18px;
-    display: flex;
-    justify-content: center;
-    > .foldout {
-      background-color: ${colors.neutralBackgroundColor};
-      color: ${colors.linkColor};
-      padding: 8px;
-      min-height: 24px;
-      min-width: 200px;
-      border-radius: 50px;
-      border: none;
+  // Reset the url when closing the dialog
+  // biome-ignore lint/correctness/useExhaustiveDependencies:
+  useEffect(() => {
+    if (
+      !dialogOpen &&
+      activeImage &&
+      window.location.pathname !== baseFeatureUrl
+    ) {
+      router.push(baseFeatureUrl);
     }
-  }
+  }, [dialogOpen]);
 
-  &.open {
-    > .images {
-    max-height: 800px;
-    overflow: scroll;
-    > .image {
-      max-height: 500px;
+  // Set the url when navigating to an image
+  // biome-ignore lint/correctness/useExhaustiveDependencies:
+  useEffect(() => {
+    if (!dialogOpen || !activeImage) {
+      return;
     }
+    const url = getDetailPageUrl(activeImage._id);
+    if (url !== window.location.pathname) {
+      router.push(url, undefined, { shallow: true });
     }
-  }
-`
+  }, [activeImage]);
+
+  // The Dialog component can autofocus on its own, when using the Dialog.Trigger component.
+  // But in this case, we want to focus the currently active image and not the one that actually
+  // opened the Dialog in the first place, so that's why this functionality is implemented a
+  // second time.
+  const galleryRef = useRef<HTMLElement>(null);
+  const handleOnCloseAutoFocus = (event: Event) => {
+    event.preventDefault();
+    galleryRef.current
+      ?.querySelector<HTMLAnchorElement>(`[data-image-id=${activeImage?._id}]`)
+      ?.focus();
+  };
+
```

Comment:
This does not seem to work for me (Firefox and Chrome). After leaving the full screen image, no element is focused on the map screen

Reply:
Shame on me, I haven't tested in Chrome and for some weird reason it worked in Firefox at some time, even though it shouldn't have 😄 It should be fixed now!

Reply:
I tested the new commits on Chrome and it works now! Thank you :)